awx_collection/plugins/modules/tower_token.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/python
+# coding: utf-8 -*-
+
+
+# (c) 2020, John Westcott IV <john.westcott.iv@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: tower_token
+author: "John Westcott IV (@john-westcott-iv)"
+version_added: "2.3"
+short_description: create, update, or destroy Ansible Tower tokens.
+description:
+    - Create or destroy Ansible Tower tokens. See
+      U(https://www.ansible.com/tower) for an overview.
```

Comment:
I feel like this is not enough detail, see your ovirt example

https://github.com/ansible/ansible/blob/stable-2.9/lib/ansible/modules/cloud/ovirt/ovirt_auth.py#L19-L20

> which should be later used in all other oVirt/RHV modules, so all modules don't need to perform login and logout.

It is implied in its wording that it's intended that you use the module, and then don't pass the token value manually, because it goes through ansible_facts. You can say any variation on this, but it needs to be said.